[dev] Reduce core zip-file size by skipping vendor tests

triple0t opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes https://github.com/woocommerce/woocommerce/issues/55967

This PR reduces the WooCommerce plugin size by removing unnecessary files, such as test files, from the vendor directory.

This will shrink the woocommerce plugin build by 0.2MB (compressed, more than 2.3MB uncompressed)

- Optimize vendor files during Composer install/update
- Add script to remove test directories from vendor folder
- Clean up email-editor package contents during installation
- Reduce plugin package size by removing unnecessary files




### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|   17.5MB (47.6MB uncompressed)     |   17.3MB (45.3MB uncompressed)    |

**Before**

![Image](https://github.com/user-attachments/assets/ef5f9a95-b467-4843-ab39-518afffdc082)

**After**

![Image](https://github.com/user-attachments/assets/c6e657dd-d188-4aa4-a8ba-e08c6143e5ad)


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Before checking out to this branch, run the `pnpm --filter='@woocommerce/plugin-woocommerce' build:zip` command
2. Copy out the final `woocommerce.zip` file and save it somewhere
3. Checkout to this branch and run the command again  (`pnpm --filter='@woocommerce/plugin-woocommerce' build:zip`)
4. Compare the size difference and notice the new zip file is smaller.

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

triple0t committed:
Optimize vendor files during Composer install/update

- Add script to remove test directories from vendor folder
- Clean up email-editor package contents during installation
- Reduce plugin package size by removing unnecessary files

triple0t committed:
Add changelog files

kalessil commented:
While it works as intended, I think we can simplify and offload those operations from the installation phase to the zip-build itself.
In `plugins/woocommerce/bin/build-zip.sh` we have [a rsync call](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/bin/build-zip.sh#L33) which filters out what's getting into the final zip-file.

Leveraging it, we can refine the optimization to the following changes (and we'll need to drop `codeception.yml` sihce the package was migrated on PHPUnit):
```patch
Subject: [PATCH] Refine zip size minimization.
---
Index: plugins/woocommerce/.distignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/woocommerce/.distignore b/plugins/woocommerce/.distignore
--- a/plugins/woocommerce/.distignore	(revision ebff36ab61173f021fa54ddd7a7fd51daf16dec2)
+++ b/plugins/woocommerce/.distignore	(date 1740759641031)
@@ -12,6 +12,10 @@
 /playwright-report/
 /storybook/
 /tests/
+/vendor/*/*/tests
+/packages/*/tests
+/packages/*/changelog
+/packages/email-editor/tasks
 /e2e/
 babel.config.js
 blocks.ini
@@ -29,11 +33,12 @@
 phpcs.xml
 phpunit.xml
 phpunit.xml.dist
+phpunit-*.xml.dist
 postcss.config.js
 project.json
 README.md
 renovate.json
 tsconfig.*
 webpack.config.js
-client/legacy/
+/client/legacy/
 /client/admin/
```

triple0t committed:
Remove vendor optimization scripts and update .distignore

- Remove `optimize-vendor-files.sh` script from WooCommerce plugin
- Update `.distignore` to exclude more test and unnecessary files
- Remove Codeception configuration for email-editor package
- Update Composer scripts to remove vendor optimization steps

triple0t commented:
Thanks for the review, @kalessil 
> While it works as intended, I think we can simplify and offload those operations from the installation phase to the zip-build itself.
In plugins/woocommerce/bin/build-zip.sh we have [a rsync call](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/bin/build-zip.sh#L33) which filters out what's getting into the final zip-file.
> 
> Leveraging it, we can refine the optimization to the following changes (and we'll need to drop codeception.yml sihce the package was migrated on PHPUnit

I agree with you. Implemented in https://github.com/woocommerce/woocommerce/pull/55971/commits/67b740986e7d60cb9756a94b28a5b46f7d0cd32f

kalessil reviewed (approved):
LGTM and thanks for wrangling this one!

The tests are no longer added to the zip file, and the bundled email editor package version contains only the necessary files (122kB zip-file size reduction).

```
-rw-rw-r--  1 ... ... 17234581 Feb 28 21:01 woocommerce-trunk.zip
-rw-rw-r--  1 ... ... 17109340 Feb 28 20:54 woocommerce-updated.zip
```
